stingray/simulator/tests/test_simulator.py

```diff
@@ -0,0 +1,117 @@
+import numpy as np
+
+from astropy.tests.helper import pytest
+from stingray.simulator import simulator, models
+from scipy.stats import chisquare
+
+class TestSimulator(object):
+
+    @classmethod
+    def setup_class(self):
+        self.simulator = simulator.Simulator(N=1024)
+
+    def test_simulate_with_seed(self):
+        """
+        Simulate with a random seed value.
+        """
+        self.simulator = simulator.Simulator(N=1024, seed=12)
+        assert len(self.simulator.simulate(2).counts), 1024
+
+    def test_simulate_powerlaw(self):
+        """
+        Simulate light curve from power law spectrum.
+        """
+        assert len(self.simulator.simulate(2).counts), 1024
+        
+    def test_compare_powerlaw(self):
+        """
+        Compare simulated power spectrum with actual one.
+        """        
+        B, N, red_noise, dt = 2, 1024, 10, 1
+
+        self.simulator = simulator.Simulator(N=N, dt=dt, mean=5, rms=1, red_noise=red_noise)
+        lc = [self.simulator.simulate(B) for i in xrange(1,30)]
```

Comment:
Instead of xrange, please use just range. It will be very efficient in Py3, less in Py2 but we are privileging Py3